delta/rdf/src/main/scala/ch/epfl/bluebrain/nexus/delta/rdf/IriOrBNode.scala

```diff
@@ -47,6 +48,51 @@ object IriOrBNode {
     */
   final case class Iri private (private val value: IRI) extends IriOrBNode {
 
+    /**
+      * Extract the query parameters as key and values
+      */
+    def query(): Query =
+      Query(Option(value.getRawQuery))
+
+    /**
+      * Extract the query parameters as String
+      */
+    def rawQuery(): String = Option(value.getRawQuery).getOrElse("")
+
+    /**
+      * Removes each encounter of the passed query parameter keys from the current Iri query parameters
+      *
+      * @param keys the keys to remove
+      */
+    def removeQueryParams(keys: String*): Iri =
+      if (rawQuery().isEmpty) this
+      else queryParams(Query(query().toMap -- keys))
+
+    /**
+      * Override the current query parameters with the passed ones
+      */
+    def queryParams(query: Query): Iri = {
+      val sb = new StringBuilder
```

Comment:
The iriFactory has methods that could help here

Reply:
The construct method does not work, it is [not implemented](https://github.com/apache/jena/blob/master/jena-iri/src/main/java/org/apache/jena/iri/IRIFactory.java#L473)